Source/JavaScriptCore/wasm/WasmTypeDefinition.cpp

```diff
@@ -158,10 +168,10 @@ static unsigned computeSignatureHash(size_t returnCount, const Type* returnTypes
 
 static unsigned computeStructTypeHash(size_t fieldCount, const FieldType* fields)
 {
-    unsigned accumulator = 0x15d2546;
+    unsigned init = 0x15d2546;
+    unsigned accumulator = init;
     for (uint32_t i = 0; i < fieldCount; ++i) {
-        accumulator = WTF::pairIntHash(accumulator, WTF::IntHash<uint8_t>::hash(static_cast<uint8_t>(fields[i].type.kind)));
-        accumulator = WTF::pairIntHash(accumulator, WTF::IntHash<uint8_t>::hash(static_cast<uint8_t>(fields[i].type.index)));
+        accumulator = WTF::pairIntHash(accumulator, fields[i].type.hash(init));
```

Comment:
I mentioned in a comment in the header for this that you might be able to simplify `hash` here. You could potentially call `kind()` and `index()` on the storage type instead of needing a helper `hash` method.